### ISSUE TITLE
"Ambiguous Type Variable" error message should say which variable is ambiguous

### ISSUE THREAD

**Issue by duvenaud** [google-research/dex-lang#1076]
This always slows me down.  For a concrete example, I was working on refactoring the `Subset` interface:

```
interface Subset a_sub a
  inject'         : a_sub -> a
  can_project     : a -> Bool
  unsafe_project' : a -> a_sub

def project {a} (a_sub:Type) [Subset a_sub a] (x:a) : Maybe a_sub =
  case can_project x of
    True  -> unsafe_project' x
    False -> Nothing
Type error:Ambiguous type variable

  case can_project x of
       ^^^^^^^^^^^^
```

It took me a while to realize that `a_sub` was the ambiguous variable here.

**Comment by dougalm** [comment]
I think this is fixed by #1259 

Updated to the new syntax, your example now looks something like this:

```
interface Subset2(a_sub, a)
  inject2'         : (a_sub) -> a
  can_project2     : (a) -> Bool
  unsafe_project2' : (a) -> a_sub

def project2(a_sub:Type, given () (Subset2 a_sub a), x:a) -> Maybe a_sub given (a) =
  case can_project2 x of
    True  -> Just $ unsafe_project2' x
    False -> Nothing
> Type error:Couldn't infer implicit argument a_sub of can_project2
>
>   case can_project2 x of
>        ^^^^^^^^^^^^^^^
```

And the fix is to provide `a_sub` explicitly, like this:
```
def project3(a_sub:Type, given () (Subset2 a_sub a), x:a) -> Maybe a_sub given (a) =
  case can_project2(a_sub=a_sub, x) of
    True  -> Just $ unsafe_project2' x
    False -> Nothing
```

**Comment by duvenaud** [comment]
Wonderful!  So satisfying to see all these issues being knocked down at once.
